sympy/physics/units/tests/test_dimensions.py

```diff
@@ -65,17 +66,23 @@ def test_Dimension_properties():
 
 
 def test_Dimension_add_sub():
-    assert length + length == length
+    assert length + length == length + foot == foot + length == length
     assert length - length == length
     assert -length == length
 
-    raises(TypeError, lambda: length + 1)
-    raises(TypeError, lambda: length - 1)
     raises(ValueError, lambda: length + time)
```

Comment:
I believe that `length + 1` should raise an exception in any dimension system, whereas `length + time` should not always, e.g. if we use natural units: https://github.com/sympy/sympy/issues/14374